### PR TITLE
[spark] Ray on spark support JSON format option values and set default object_spilling_config path 

### PR DESCRIPTION
**Pull request by WeichenXu123**
## Why are these changes needed?

For Ray on spark:
* support JSON format option values
* set default object_spilling_config path , this is specifically for databricks runtime, it sets spilling path to path under "/local_disk0" by default to aviod disk space overflow.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — COMMENTED** [review]
Could you add a test?

File: python/ray/util/spark/cluster_init.py [hunk 1]
```diff
@@ -485,7 +502,9 @@ def _setup_ray_cluster(
         f"--memory={heap_memory_head_node}",
         f"--object-store-memory={object_store_memory_head_node}",
         *dashboard_options,
-        *_convert_ray_node_options(head_node_options),
+        *_convert_ray_node_options(
+            head_node_options, object_spilling_path=object_spilling_dir
```
[the remark this answers is not in the document]
Comment (WeichenXu123) [reply python/ray/util/spark/cluster_init.py:506]: according to ray doc:
https://docs.ray.io/en/latest/ray-core/objects/object-spilling.html#cluster-mode

> # You only need to specify the config when starting the head node, all the worker nodes will get the same config from the head node.

File: python/ray/util/spark/cluster_init.py [hunk 2]
```diff
@@ -219,15 +220,29 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+        if "system_config" not in options:
+            options["system_config"] = {}
+        sys_conf = options["system_config"]
+        if "object_spilling_config" not in sys_conf:
+            sys_conf["object_spilling_config"] = json.dumps({
+              "type": "filesystem",
+              "params": {
+                "directory_path": object_spilling_path,
+              }
+            })
```
[the remark this answers is not in the document]
Comment (WeichenXu123) [reply python/ray/util/spark/cluster_init.py:243]: agreed!

**Comment by WeichenXu123** [comment]
@jjyao Done!

File: python/ray/util/spark/cluster_init.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjyao) [reply python/ray/util/spark/cluster_init.py]: Oh yea. Forgot that.

**Review by jjyao — APPROVED** [review]

**Pull request merged**